File: Problems on N numbers/Program12.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>

/////////////////////////////////////////////////////////////////////
//
//  Function Name:  SearchLastOccurance
//  Description :   Used to return Last Occurence of Element
//  Input :         Integer Array,Integer,Integer
//  Output :        Integer
//  Date :          4-May-2022
//  Author :        Abhishek Balasaheb Mandalik
//
/////////////////////////////////////////////////////////////////////

int SearchLastOccurance(int Arr[], int iLength, int iNo)
{
    int iCnt = 0;

    for(iCnt = iLength-1; iCnt >= 0; iCnt--)
    {
            if(Arr[iCnt] == iNo)      
            {
                break;
            }
    }
        return iCnt;
}

/////////////////////////////////////////////////////////////////////
//
//  Function Name:  Main
//  Description :   Entry Point Function of the Application
//  Date :          4-May-2023
//  Author :        Abhishek Balasaheb Mandalik
//
/////////////////////////////////////////////////////////////////////

int main()
{
    int iSize = 0, iCnt = 0, iValue = 0, iRet;
    int *ptr = NULL;

    printf("Enter number of elements\n");
    scanf("%d",&iSize);

    ptr = (int *)malloc(sizeof(int) * iSize);

    printf("Enter the values\n");
    for(iCnt = 0; iCnt < iSize; iCnt++)
    {
        scanf("%d",&ptr[iCnt]);
    }

    printf("Enter the element to search\n");
    scanf("%d",&iValue);

    iRet = SearchLastOccurance(ptr, iSize, iValue);
    if(iRet == -1)
    {
        printf("There is no such element in array\n");
    }
    else
    {
        printf("Element last occured at : %d\n",iRet);
    }
     free(ptr);

    return 0;
}

File: Problems on Bitwise/Program12.java
import java.lang.*;
import java.util.*;

//              0   0   1   0   1   0   1   1       iNo
//              0   0   0   0   1   0   0   0       iMask
//-------------------------------------------
//              0   0    1  0   0   0   1   1       iResult


//              0   0   1   0   0   0   1   1       iNo
//              0   0   0   0   1   0   0   0       iMask
//-------------------------------------------
//              0   0   1   0   1   0   1   1       iReult


//  0000    0000    0000    0000    0010    0000    0000    0000
//      0       0           0         0          2          0           0       0


/////////////////////////////////////////////////////////////////////
//
//  Class Name:	Bitwise 
//
//	Function Name:	ToggleBit
//	Description :   Used to Toggle Bit
//  	Input :		Integer,Integer
//  	Output :	Integer
//  	Date :		13-June-2022
//
//  Author :	Abhishek Balasaheb Mandalik
//
/////////////////////////////////////////////////////////////////////

class Bitwise
{
        public int ToggleBit(int iNo, int iPos)
        {
            if((iPos <=0) || (iPos > 32))
            {
                System.out.println("Invalid position");
                return 0;
            }

            int iMask = 0X00000001;
            int iResult = 0;

            iMask = iMask << (iPos-1);

            iResult = iNo ^ iMask;
            return iResult;
        }
}

/////////////////////////////////////////////////////////////////////
//
//  Class Name:	Program12 
//
//	Function Name:  Main
//	Description :   Entry Point Function of the Application
//	Date :          13-June-2023
//
//  Author :	Abhishek Balasaheb Mandalik
//
/////////////////////////////////////////////////////////////////////

class Program12
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);  // 9th July

        System.out.println("Enter number ");
        int value = sobj.nextInt();

        System.out.println("Enter position ");
        int pos = sobj.nextInt();

        Bitwise bobj = new Bitwise();
        int iRet = bobj.ToggleBit(value,pos);
        System.out.println("Updated number is : "+iRet);
    }
}


